### PR TITLE
overlays: implement native send/recv message dialogs

### PR DESCRIPTION
**Pull request by Megamouse**
- implements a native version of the recvmessage dialog (netplay invites etc)
- implements a native version of the sendmessage dialog (netplay invites etc)
- adds more error checks to the respective sceNp functions
- hooks up gui for sceNpBasicAddFriend
- hooks up gui for sceNpBasicSendMessageAttachment
- replaces vblank in overlay update with timestamp in order to fix fade animations when rsx is idle

TODO:
- [x] Fix softlock when dialog is opened (see #15123)

image
image

### CODE REVIEW

**Comment by Darkhost1999** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/14072767/RPCS3.log)
Closes to desktop on recvmessage dialog

**Comment by Megamouse** [comment]
Yes. I forgot to initialize something. Try again please

**Comment by Darkhost1999** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/14073442/RPCS3.log)
Same result. Close to desktop.

**Comment by Megamouse** [comment]
It works now. But there's still a bug that occasionally freezes the game when the dialog is supposed to open.

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 1]
```diff
@@ -1410,40 +1404,23 @@ error_code sceNpBasicSendMessageGui(ppu_thread& ppu, vm::cptr<SceNpBasicMessageD
 	return CELL_OK;
 }
 
-error_code sceNpBasicSendMessageAttachment(vm::cptr<SceNpId> to, vm::cptr<char> subject, vm::cptr<char> body, vm::cptr<char> data, u32 size, sys_memory_container_t containerId)
+error_code sceNpBasicSendMessageAttachment(ppu_thread& ppu, vm::cptr<SceNpId> to, vm::cptr<char> subject, vm::cptr<char> body, vm::cptr<void> data, u32 size, sys_memory_container_t containerId)
 {
 	sceNp.todo("sceNpBasicSendMessageAttachment(to=*0x%x, subject=%s, body=%s, data=%s, size=%d, containerId=%d)", to, subject, body, data, size, containerId);
```
Comment (RipleyTom) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:1409]: .warning

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 2]
```diff
@@ -1657,40 +1634,23 @@ error_code sceNpBasicAbortGui()
 	return CELL_OK;
 }
 
-error_code sceNpBasicAddFriend(vm::cptr<SceNpId> contact, vm::cptr<char> body, sys_memory_container_t containerId)
+error_code sceNpBasicAddFriend(ppu_thread& ppu, vm::cptr<SceNpId> contact, vm::cptr<char> body, sys_memory_container_t containerId)
 {
 	sceNp.todo("sceNpBasicAddFriend(contact=*0x%x, body=%s, containerId=%d)", contact, body, containerId);
```
Comment (RipleyTom) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:1639]: .warning

**Pull request merged**